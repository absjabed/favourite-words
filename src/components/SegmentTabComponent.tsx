import React from 'react';
import {Pressable, StyleSheet, View} from 'react-native';
import Colors from '../constants/Colors/Colors';
import Row from '../screens/Layout/Row';
import AppText from './Text';


type Props = {
    isOnSearch: boolean;
    changeView: () => void;
  };

const SegmentTab = ({isOnSearch, changeView}: Props) => {
  return (
    <Row style={styles.container}>
        <Pressable style={{...styles.searchButtonStyle, backgroundColor: isOnSearch ? Colors.primary : Colors.primaryLight }} onPress={()=> changeView()}>
            <AppText style={styles.text}>
                Search Dictionary
            </AppText>
        </Pressable>
        <Pressable style={{...styles.favouriteButtonStyle, backgroundColor: isOnSearch ? Colors.primaryLight : Colors.primary}} onPress={()=> changeView()}>
            <AppText style={styles.text}>
                View Favourites
            </AppText>
        </Pressable>
    </Row>
  );
};

export default SegmentTab;

const styles = StyleSheet.create({
  container: {
    paddingTop: 10, 
    justifyContent:'center',
    backgroundColor: '#fff'
  },
  searchButtonStyle:{
    padding:10, 
    borderTopLeftRadius: 7, 
    borderBottomLeftRadius: 7, 
    backgroundColor: Colors.primary
  },
  favouriteButtonStyle:{
    padding:10,
    borderTopRightRadius: 7, 
    borderBottomRightRadius: 7,  
    backgroundColor: Colors.primaryLight
  },
  text: {
    fontSize: 13,
    color: '#fff',
    fontFamily:'Ubuntu-Regular',
    justifyContent: 'center',
  },
});
