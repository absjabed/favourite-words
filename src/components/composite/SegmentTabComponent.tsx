import React from 'react';
import {Pressable, StyleSheet} from 'react-native';
import Colors from '../../constants/Colors/Colors';
import Row from '../../screens/Layout/Row';
import AppText from '../Text';


type Props = {
    isOnSearch: boolean;
    changeView: () => void;
  };

const SegmentTab = ({isOnSearch, changeView}: Props) => {
  return (
    <Row style={styles.container}>
        <Pressable style={{...styles.searchButtonStyle,  backgroundColor: isOnSearch ? Colors.activeColor : Colors.deactiveColor}} onPress={()=> changeView()}>
            <AppText style={{...styles.text, color: isOnSearch ? Colors.white : Colors.placeholderColor, fontWeight: isOnSearch ? '600' : '100'}}>
                Search Dictionary
            </AppText>
        </Pressable>
        <Pressable style={{...styles.favouriteButtonStyle, backgroundColor: isOnSearch ? Colors.deactiveColor : Colors.activeColor}} onPress={()=> changeView()}>
            <AppText style={{...styles.text, color: isOnSearch ? Colors.placeholderColor : Colors.white, fontWeight: isOnSearch ? '100' : '600'}}>
                View Favourites
            </AppText>
        </Pressable>
    </Row>
  );
};

export default SegmentTab;

const styles = StyleSheet.create({
  container: {
    paddingTop: 20, 
    justifyContent:'center',
    backgroundColor: Colors.darkBackground
  },
  searchButtonStyle:{
    padding:10, 
    borderTopLeftRadius: 7, 
    borderBottomLeftRadius: 7, 
    borderLeftColor: Colors.grey, 
    borderTopColor: Colors.grey, 
    borderBottomColor: Colors.grey, 
    borderWidth: 0.8 
  },
  favouriteButtonStyle:{
    padding:10,
    borderTopRightRadius: 7, 
    borderBottomRightRadius: 7,
    borderRightColor: Colors.grey, 
    borderTopColor: Colors.grey, 
    borderBottomColor: Colors.grey, 
    borderWidth: 0.8
  },
  text: {
    fontSize: 13,
    color: Colors.white,
    fontFamily:'Ubuntu-Regular',
    justifyContent: 'center',
  },
});
