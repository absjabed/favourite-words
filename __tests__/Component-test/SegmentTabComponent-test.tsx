import 'react-native';
import React from 'react';
import SegmentTabComponent from '../../src/components/composite/SegmentTabComponent';
import renderer, { create } from 'react-test-renderer';

const mockIsOnSearchView = true,
  mockChangeViewFunc = () => console.log("view changed");

it('SegmentTabComponent renders correctly', () => {
  renderer.create(<SegmentTabComponent isOnSearch={mockIsOnSearchView} changeView={mockChangeViewFunc}/>);
});

test('SegmentTabComponent snapshot', () => {
  const tree = create(<SegmentTabComponent isOnSearch={mockIsOnSearchView} changeView={mockChangeViewFunc}/>);
  expect(tree).toMatchSnapshot();
});
