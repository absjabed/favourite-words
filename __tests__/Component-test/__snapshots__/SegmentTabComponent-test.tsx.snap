// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentTabComponent snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      },
      Object {
        "backgroundColor": "#121212",
        "justifyContent": "center",
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0d0d0d",
        "borderBottomColor": "#acacac",
        "borderBottomLeftRadius": 7,
        "borderLeftColor": "#acacac",
        "borderTopColor": "#acacac",
        "borderTopLeftRadius": 7,
        "borderWidth": 0.8,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Ubuntu-Regular",
          "fontSize": 13,
          "fontWeight": "600",
          "justifyContent": "center",
        }
      }
    >
      Search Dictionary
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#212121",
        "borderBottomColor": "#acacac",
        "borderBottomRightRadius": 7,
        "borderRightColor": "#acacac",
        "borderTopColor": "#acacac",
        "borderTopRightRadius": 7,
        "borderWidth": 0.8,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#9b9b9b",
          "fontFamily": "Ubuntu-Regular",
          "fontSize": 13,
          "fontWeight": "100",
          "justifyContent": "center",
        }
      }
    >
      View Favourites
    </Text>
  </View>
</View>
`;
